cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -132,14 +140,40 @@ func (o *objectGraph) addObj(obj *unstructured.Unstructured) {
 	// Adds the node to the Graph.
 	newNode := o.objToNode(obj)
 
-	// Process OwnerReferences; if the owner object doe not exists yet, create a virtual node as a placeholder for it.
+	// Process OwnerReferences; if the owner object does not exists yet, create a virtual node as a placeholder for it.
+	o.processOwnerReferences(obj, newNode)
+}
+
+// addRestoredObj adds a Kubernetes object to the object graph from file that is generated during a restore
+// Populates the restoredObject field to be referenced during restore
+// During add, OwnerReferences are processed in order to create the dependency graph.
+func (o *objectGraph) addRestoredObj(obj *unstructured.Unstructured) error {
+	// Add object to graph
+	o.objToNode(obj)
+
+	// Check to ensure node has been added to graph
+	node, found := o.uidToNode[obj.GetUID()]
+	if !found {
+		return errors.Errorf("error adding obj %v with id %v to object graph", obj.GetName(), obj.GetUID())
+	}
+
+	// Copy the raw object yaml to be referenced when restoring object
+	node.restoreObject = obj.DeepCopy()
+
+	// Process OwnerReferences; if the owner object does not exists yet, create a virtual node as a placeholder for it.
+	o.processOwnerReferences(obj, node)
```

Comment:
Hmm, `processOwnerReferences` isn't getting called inside the `objToNode` func. And I don't think calling `processOwnerReferences` will work inside the `objToNode` func since we may return early if an existing node is already found. I arbitraily tried to move `processOwnerReferences` to the end of `objToNode` but it had a number of failures in the object graph testing suite

Reply:
You are right. sorry I got confused